Index: working_jdktools/modules/samsa/src/main/native/samsa/samsa.c
===================================================================
--- working_jdktools/modules/samsa/src/main/native/samsa/samsa.c	(revision 808406)
+++ working_jdktools/modules/samsa/src/main/native/samsa/samsa.c	(working copy)
@@ -149,6 +149,10 @@
         return 1;
     }
     
+    char *debian_suffix = strstr(toolName, "-harmony");
+    if (debian_suffix != NULL)
+      *debian_suffix = '\0';
+
     isJavaw = strcmp(toolName, "javaw") == 0;
 
     toolType = getToolType(root, toolName);
